From 4b337793e8fb21ecdffc35e7d7dec75bf9388cae Mon Sep 17 00:00:00 2001
From: Richard Nelson <wabz@pidgin.im>
Date: Thu, 12 Oct 2006 01:02:07 +0000
Subject: [PATCH] [gaim-migrate @ 17462] Patch #1574229 from Richard Nelson
 (wabz) to fix a crash when adding a buddy and show-offline
 is set to true.

committer: Sadrul Habib Chowdhury <sadrul@pidgin.im>

--HG--
extra : convert_revision : d19967f6089487f9dd8cce827daa8f7784fec426
---
 console/gntblist.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/console/gntblist.c b/console/gntblist.c
index a9d8294..f63e559 100644
--- a/console/gntblist.c
+++ b/console/gntblist.c
@@ -206,6 +206,8 @@ node_update(GaimBuddyList *list, GaimBlistNode *node)
 		if ((!gaim_prefs_get_bool(PREF_ROOT "/showoffline") && !is_contact_online(contact)) ||
 				contact->currentsize < 1)
 			node_remove(gaim_get_blist(), node);
+		else
+			add_node(node, list->ui_data);
 	} else if (GAIM_BLIST_NODE_IS_GROUP(node)) {
 		GaimGroup *group = (GaimGroup*)node;
 		if ((!gaim_prefs_get_bool(PREF_ROOT "/showoffline") && !is_group_online(group)) ||
@@ -526,6 +528,8 @@ add_buddy(GaimBuddy *buddy, GGBlist *ggblist)
 		return;
 
 	contact = (GaimContact*)node->parent;
+	if (!contact)   /* When a new buddy is added and show-offline is set */
+		return;
 	add_node((GaimBlistNode*)contact, ggblist);
 
 	node->ui_data = gnt_tree_add_row_after(GNT_TREE(ggblist->tree), buddy,
-- 
1.7.10.4

